homework/CSharpAdvanced/Program.cs

```diff
@@ -0,0 +1,95 @@
+﻿using CSharpAdvanced;
+using CSharpOOP2;
+
+#region Task_1
+
+GenericMetod genericMetod = new GenericMetod();
+
+int[] intItems = new int[] { 1, 2, 3, 4, 5 };
+double[] doubleItems = new double[] { 1.1, 2, 2, 3.3, 4.4, 5.5 };
+string[] stringItems = new string[] { "Hello", "world" };
+
+genericMetod.GetFirstItem(intItems);
+genericMetod.GetFirstItem(doubleItems);
+genericMetod.GetFirstItem(stringItems);
+
+#endregion
+
+#region Task_2
+
+Rose rose = new Rose("Rose", "Black", 10);
+Chamomile chamomile = new Chamomile("Chamomile", "White", 5);
+
+Console.WriteLine("Enter the number of rose flowers:");
+int numberOfRose = int.Parse(Console.ReadLine());
+
+Console.WriteLine("Enter the number of chamomile flowers:");
+int numberOfChamomile = int.Parse(Console.ReadLine());
+
+Console.WriteLine($"Final chek:");
+
+Console.WriteLine(StringCheck(rose));
+Console.WriteLine(StringCheck(chamomile));
+
+Console.WriteLine($"Total payable: {FinalCheck()}");
+
+string StringCheck<T>(T flower) where T : Flower
+{
+    return $"Name: {flower.Name}; Color: {flower.Color}; Price: {flower.Price};" +
+        $"BouquetPrice: {GetPrice(flower, numberOfRose)}";
+}
+
+double GetPrice<T>(T flower, int number) where T : Flower
+{
+    return flower.Price * number;
+}
+
+double FinalCheck()
```

Comment:
same :)
why does this method have no params, but is used with variables inside? If I will decide to move it into helper class?
It you want to print final tip for all flowers you cal pass dictionary to method where Key is Flower, Value is its number :)